app/src/androidTest/java/org/simple/clinic/bloodsugar/BloodSugarRepositoryAndroidTest.kt

```diff
@@ -48,22 +48,25 @@ class BloodSugarRepositoryAndroidTest {
 
   @Test
   fun saving_a_blood_sugar_reading_should_work_correctly() {
-    //given
-    val bloodSugarReading = BloodSugarReading(value = 10, type = Random)
+    // given
+    val now = Instant.now(clock)
+    val bloodSugarReading = BloodSugarReading(value = 50, type = Random)
     val patientUuid = UUID.fromString("a5921ec9-5c70-421a-bb0b-1291364683f6")
     val user = testData.qaUser()
+    val bloodSugarUuid = UUID.fromString("e35dea7a-b13b-4ab1-aaa4-0c5998e5c79f")
 
-    //when
-    val testObserver = repository.saveMeasurement(
+    // when
+    val savedBloodSugar = repository.saveMeasurement(
+        uuid = bloodSugarUuid,
         reading = bloodSugarReading,
         patientUuid = patientUuid,
         loggedInUser = user,
         facility = testData.qaFacility(),
-        recordedAt = Instant.now(clock)
-    ).test()
+        recordedAt = now)
+        .blockingGet()
 
-    //then
-    testObserver.assertComplete().assertNoErrors()
+    // then
+    assertThat(savedBloodSugar.uuid).isEqualTo(bloodSugarUuid)
```

Comment:
We should verify the entire `BloodSugarMeasurement` model here and not just the UUID.